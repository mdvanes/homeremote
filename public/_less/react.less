// for /react test

// Toggle
.btn-icon {
    font-size: 350%;
}

// Button-Set
.btn-group-justified {
    > a,
    > span {
        font-size: 230%;
    }
}

.margin-top {
    margin-top: 1em;
}

.log {
    color: #fff;
    height: 300px;
}