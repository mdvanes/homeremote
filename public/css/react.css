.btn-icon {
  font-size: 350%;
}
.btn-group-justified > a,
.btn-group-justified > span {
  font-size: 230%;
}
.margin-top {
  margin-top: 1em;
}
.log {
  color: #fff;
  height: 300px;
}
