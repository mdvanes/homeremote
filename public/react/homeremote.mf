CACHE MANIFEST
index.html
../library/bootstrap/css/bootstrap.min.css
../library/bootstrap/fonts/glyphicons-halflings-regular.woff2
../library/bootstrap/fonts/glyphicons-halflings-regular.woff
../library/bootstrap/fonts/glyphicons-halflings-regular.ttf
../css/react.css
build/bundle.js

NETWORK:
*
