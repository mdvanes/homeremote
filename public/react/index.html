<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>React</title>
    
    
  </head>
  <body>
    
    <script src="build/bundle.js"></script>
    
  </body>
</html>
