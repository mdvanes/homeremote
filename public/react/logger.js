/**
 * Created by m.van.es on 6-2-2016.
 */

let logger = {};

let writeLog = (...messages) => {
    let time = new Date();
    let timestamp = time.getHours() + ':' +
        time.getMinutes() + '.' +
        time.getSeconds();
    document.querySelector('.log').value = timestamp + ' ' +
        messages.join(' ') + '\n' +
        document.querySelector('.log').value;
};

logger.log = message => {
    writeLog('INFO:    ', message);
};

logger.error = message => {
    writeLog('ERROR:', message);
};

export default logger;