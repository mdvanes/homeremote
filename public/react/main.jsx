// from https://github.com/code0wl/react-example-es2015/blob/master/app/main.jsx#L5
//import './stylesheets/main.css';
import React from 'react';
import {render} from 'react-dom';
////import FilterableProductTable from './src/filterable-product-table';
import ButtonGroup from './components/button-group';
import Toggle from './components/toggle';
import Log from './components/log';

// init shell
renderShell();

// TODO toggle as single button with icon (instead of 2 buttons)

function renderShell() {
    var shell = document.createElement('div');
    shell.className = 'app-shell container';
    document.body.appendChild(shell);
    render(
        <div>
            <div className="row">
                <div className="col-xs-12">
                    <ButtonGroup label=" 1" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 2" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 3" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 4" icon="lamp" id="clickstub"/>
                </div>
            </div>
            <div className="row margin-top">
                {/*<Toggle label=" 2" id="togglestub" icon="lamp"/>
                <Toggle label=" 3" id="togglestub" icon="lamp"/>
                <Toggle label="Switch 4" id="togglestub"/>*/}
                <Toggle label="" id="togglestub" icon="volume-up"/>{/* Broadcast */}
                <Toggle label="" id="togglestub" icon="music"/>{/* Music on server */}
            </div>
            <div className="row margin-top">
                <div className="col-xs-12">
                    <Log/>
                </div>
            </div>
        </div>, shell);
}

