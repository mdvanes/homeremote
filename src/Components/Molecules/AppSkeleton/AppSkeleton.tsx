import React, { FC } from "react";
import theme from "../../../theme";
import {
    AppBar,
    Box,
    Container,
    Grid,
    IconButton,
    MuiThemeProvider,
    Toolbar,
    Typography,
} from "@material-ui/core";
import { Skeleton } from "@material-ui/lab";
import MenuIcon from "@material-ui/icons/Menu";

const AppSkeleton: FC = () => (
    <MuiThemeProvider theme={theme}>
        <AppBar position="static">
            <Toolbar>
                <IconButton edge="start" color="inherit" aria-label="menu">
                    <MenuIcon />
                </IconButton>
                <Typography variant="h6">HomeRemote</Typography>
            </Toolbar>
        </AppBar>
        <Container maxWidth="xl">
            <Box mt={3}>
                <Grid container spacing={2}>
                    <Grid item xs={12} md={3}>
                        <Skeleton variant="rect" height={300} />
                    </Grid>
                    <Grid item xs={12} md>
                        <Skeleton variant="rect" height={300} />
                    </Grid>
                    <Grid item xs={12} md={2}>
                        <Skeleton variant="rect" height={300} />
                    </Grid>
                    <Grid item xs={12} md={2}>
                        <Skeleton variant="rect" height={300} />
                    </Grid>
                </Grid>
            </Box>
        </Container>
    </MuiThemeProvider>
);

export default AppSkeleton;
