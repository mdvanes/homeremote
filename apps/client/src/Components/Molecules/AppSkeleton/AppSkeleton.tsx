import MenuIcon from "@mui/icons-material/Menu";
import {
    AppBar,
    Box,
    Container,
    Grid,
    IconButton,
    ThemeProvider as MuiThemeProvider,
    Skeleton,
    StyledEngineProvider,
    Toolbar,
    Typography,
} from "@mui/material";
import { FC } from "react";
import createThemeWithMode from "../../../theme";

const AppSkeleton: FC = () => (
    <StyledEngineProvider injectFirst>
        <MuiThemeProvider theme={createThemeWithMode("dark")}>
            <AppBar position="static">
                <Toolbar>
                    <IconButton
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        size="large"
                    >
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6">HomeRemote</Typography>
                </Toolbar>
            </AppBar>
            <Container maxWidth="xl">
                <Box mt={3}>
                    <Grid container spacing={2}>
                        <Grid item xs={12} md={3}>
                            <Skeleton variant="rectangular" height={300} />
                        </Grid>
                        <Grid item xs={12} md>
                            <Skeleton variant="rectangular" height={300} />
                        </Grid>
                        <Grid item xs={12} md={2}>
                            <Skeleton variant="rectangular" height={300} />
                        </Grid>
                        <Grid item xs={12} md={2}>
                            <Skeleton variant="rectangular" height={300} />
                        </Grid>
                    </Grid>
                </Box>
            </Container>
        </MuiThemeProvider>
    </StyledEngineProvider>
);

export default AppSkeleton;
