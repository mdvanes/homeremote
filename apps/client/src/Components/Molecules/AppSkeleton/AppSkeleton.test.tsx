import { render } from "@testing-library/react";
import AppSkeleton from "./AppSkeleton";

describe("AppSkeleton", () => {
    it("shows a preview of the application", async () => {
        const { asFragment } = render(<AppSkeleton />);
        expect(asFragment()).toMatchSnapshot();
    });
});
