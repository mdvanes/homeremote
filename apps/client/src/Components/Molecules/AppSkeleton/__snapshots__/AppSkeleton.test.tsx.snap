// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSkeleton shows a preview of the application 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-123cs5f-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1l8lmus-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-flkwzx-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        HomeRemote
      </h6>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-178yklu"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1ev8e4z-MuiSkeleton-root"
            style="height: 300px;"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-true css-1p5zwaw-MuiGrid-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1ev8e4z-MuiSkeleton-root"
            style="height: 300px;"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 css-12xeub0-MuiGrid-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1ev8e4z-MuiSkeleton-root"
            style="height: 300px;"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 css-12xeub0-MuiGrid-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1ev8e4z-MuiSkeleton-root"
            style="height: 300px;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
